common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -131,21 +123,40 @@
         Uses the Django template engine because our webpack loader only provides template tags for Jinja and Django.
     </%doc>
     <%
-        from django.template import Template, Context
-        from webpack_loader.exceptions import WebpackLoaderBadStatsError
-        return Template("""
-            {% load render_bundle from webpack_loader %}
-            {% render_bundle entry %}
-            {% if body %}
-              <script type="text/javascript">
-                  {% autoescape off %}{{ body }}{% endautoescape %}
-              </script>
-            {% endif %}
-        """).render(Context({
-            'entry': entry,
-            'body': capture(caller.body)
-        }))
+        body = capture(caller.body)
     %>
+    ${render_bundle(entry) | n}
+    % if body:
+      <script type="text/javascript">
+        ${body | n}
+      </script>
+    % endif
+</%def>
+
+<%def name="renderReact(component, props={})">
+    <%doc>
+        Wrapper function to load a React component via webpack() and render
+        it onto the page, passing an optional context object via props.
+    </%doc>
+    <%
+        import uuid
+        id = 'react-container-{0}'.format(uuid.uuid4())
```

Comment:
Nits:

1. `id` is a builtin function in Python, and we should try to avoid shadowing builtins if possible.
2. Maybe add a comment about why we need to randomly generate the ID (as opposed to taking it as an argument for instance)?

Reply:
I randomly generated the id to force uniqueness (in case there's more than one of the same component on a page) and keep the def as simple as possible, but I'd definitely be open to taking it as an optional argument. I can see how this might be useful for tests.

Reply:
It might be simpler to force callers to deal with the duplication question when it happens. As you say, having stable names will make it simpler for tests, but probably also for debugging as a whole...?

Reply:
Yeah, that is a good point. `ReactRenderer` throws if there's more than one container with the same id anyway, so they won't get very far with duplicates.